includes/functions.php

```diff
@@ -2294,4 +2319,111 @@ function tsml_date_localised($format, $timestamp = null)
 	$datetime->setTimezone(new DateTimeZone($timezone));
 	return $datetime->format($format);
 }
+
+
+//function:	return array of updated feed records only when difference detected between matched feed record and the local db record
+function tsml_get_import_changes_only($feed_meetings, $data_source_url, $data_source_last_update, &$db_ids_to_delete, &$message_lines)
+{
+	$db_meetings = $feed_slugs = $meetings_updated = [];
+	$week_days	= [
+		__('Sunday', '12-step-meeting-list'),
+		__('Monday', '12-step-meeting-list'),
+		__('Tuesday', '12-step-meeting-list'),
+		__('Wednesday', '12-step-meeting-list'),
+		__('Thursday', '12-step-meeting-list'),
+		__('Friday', '12-step-meeting-list'),
+		__('Saturday', '12-step-meeting-list'),
+	];
+
+	// get local meetings 
+	$all_db_meetings = tsml_get_meetings();
+	$ds_ids = tsml_get_data_source_ids($data_source_url);
+	sort($ds_ids);
+
+	/* filter out all but the data source meetings  */
+	foreach ($all_db_meetings as $db_meeting) {
+		$db_id = $db_meeting['id'];
+		if (in_array($db_id, $ds_ids)) {
+			array_push($db_meetings, $db_meeting);
+		}
+	}
+
+	// create array of database slugs for matching
+	$db_slugs = array_column($db_meetings, 'slug', 'id');
+
+	// list changed and new meetings found in the data source feed
+	foreach ($feed_meetings as $meeting) {
+
+		list($day_of_week, $dow_number) = tsml_get_day_of_week_info($meeting['day'], $week_days);
+		$meeting_slug =  $meeting['slug'];
+
+		// match feed/database on unique slug
+		$is_matched = in_array($meeting_slug, $db_slugs);
+
+		if (!$is_matched) {
+			// numeric slugs may need some reformatting
+			if (is_numeric($meeting_slug)) {
+				$meeting_slug .= '-' . $dow_number;
+			}
+			$is_matched = in_array($meeting_slug, $db_slugs);
+		}
+
+		// add slug to feed array to help determine current db removals later on...
+		$feed_slugs[] = $meeting_slug;
+
+		// has the meeting been updated since the last update?
+		$current_meeting_localised_last_update = tsml_date_localised(get_option('date_format') . ' ' . get_option('time_format'), strtotime($meeting['updated']));
+		$current_meeting_last_update = strtotime($current_meeting_localised_last_update);
+		if ($current_meeting_last_update > $data_source_last_update) {
+			$meeting_name = $meeting['name']; 
+			$meeting_update_date = date('M j, Y  g:i a', $current_meeting_last_update);
```

Comment:
since this date is for output to the user, this actually seems like a place where the site date format option would be appropriate

Reply:
Good Idea...

Reply:
Get option has been implemented pretty much anywhere I could find that we are doing datetime formatting.

Reply:
Done